Task/python/Taskfile.yaml

```
@@ -0,0 +1,134 @@
+# https://taskfile.dev
+
+version: '3'
+
+includes:
+  base: 
+    taskfile: ../Taskfile.yaml
+    internal: true
+
+vars:
+  IMAGE_NAME: null
+  PYTHON_VERSION: null
+  PROJEC_SLUG: null
+  REPO_TAGS:
+    sh: git tag -l
+
+tasks:
+  lint:
+    preconditions:
+      - sh: test `git status -s | wc -w` = "0"
+        msg: "There are untracked files, please commit before linting."
+    cmds:
+      - docker run --rm -v "${PWD}:/goat" -e RUN_LOCAL=true seiso/goat:latest
+
+  test:
+    preconditions:
+      - sh: test `git symbolic-ref -q HEAD`
+        msg: "In detached HEAD state."
+    cmds:
+      - pipenv run pytest tests
+
+  build:
+    vars:
+      VERSION:
+        sh: pipenv run python -c 'from {{.PROJECT_SLUG}} import __version__; print(__version__)'
+      TAG_COMMIT_HASH:
+        sh: git rev-list -1 v{{.VERSION}}
+      COMMIT_HASH:
+        sh: git rev-parse HEAD
+      COMMIT_HASH_SHORT:
+        sh: git rev-parse --short HEAD
+      BUILD_VERSION:
+        sh: |
+          pipenv run python -c '
+          version_string = f"v{{.VERSION}}"
+          repo_tags = []
+          {{range $tag := .REPO_TAGS | splitLines -}}
+          repo_tags.append("{{$tag}}")
+          {{end}}
+          if (version_string in repo_tags and "{{.TAG_COMMIT_HASH}}" == "{{.COMMIT_HASH}}"):
+              build_version = "{{.VERSION}}"
+          else:
+              build_version = f"{{.VERSION}}-{{.COMMIT_HASH_SHORT}}"
+          print(build_version)'
+    cmds:
+      - docker buildx build --tag {{.IMAGE_NAME}}:latest --tag {{.IMAGE_NAME}}:{{.BUILD_VERSION}} .
+
+  reformat:
+    cmds:
```

Not sure if this syntax is right.  I assume if the `sh` fails, then the msg is printed and it doesn't move to the `cmds`.
```suggestion
    preconditions:
      - sh: docker pull seiso/goat:latest
        msg: "Unable to pull the latest version of the goat."
    cmds:
```

---

Or:
```suggestion
    preconditions:
      - sh: docker pull seiso/goat:latest || { echo "Failed to pull the latest goat, continuing anyway..." ;}
        msg: "Unable to pull the latest version of the goat."
    cmds:
```